Dockerfile

```diff
@@ -1,22 +1,32 @@
-FROM golang:1.11.4-alpine3.8 AS builder
-LABEL maintainer="joona@kuori.org"
+FROM golang:1.12.0-alpine3.9 AS builder
+
+ENV ACME_DNS_VERSION=v0.7.2
+
+WORKDIR /go/src/github.com/joohoi
 
-RUN apk add --update gcc musl-dev git
+RUN apk add --no-cache gcc musl-dev git && \
+    git clone -b ${ACME_DNS_VERSION} --depth 1 https://github.com/joohoi/acme-dns.git acme-dns
 
-RUN go get github.com/joohoi/acme-dns
 WORKDIR /go/src/github.com/joohoi/acme-dns
-RUN CGO_ENABLED=1 go build
 
-FROM alpine:latest
+RUN CGO_ENABLED=1 go build -ldflags="-s -w -linkmode external -extldflags -static"
 
-WORKDIR /root/
-COPY --from=builder /go/src/github.com/joohoi/acme-dns .
-RUN mkdir -p /etc/acme-dns
-RUN mkdir -p /var/lib/acme-dns
-RUN rm -rf ./config.cfg
-RUN apk --no-cache add ca-certificates && update-ca-certificates
+RUN mkdir -p /build/etc/acme-dns && \
+    mkdir -p /build/var/lib/acme-dns && \
+    cp /go/src/github.com/joohoi/acme-dns/acme-dns /build && \
+    cp /go/src/github.com/joohoi/acme-dns/config.cfg /build/etc/acme-dns && \
+    sed -Ei -e "s/(port ?= ?)\"80\"/\1\"8080\"/" \
+            -e "s/(listen ?= ?)\".*:53\"/\1\":5353\"/" /build/etc/acme-dns/config.cfg && \
+    chown 1000:0 /build/etc/acme-dns /build/var/lib/acme-dns && \
+    chmod ug+w /build/etc/acme-dns /build/var/lib/acme-dns && \
+    chmod +x /build/acme-dns
 
+
+FROM scratch
+
+LABEL maintainer="joona@kuori.org"
+COPY --from=builder /build /
+USER 1000:0
+EXPOSE 5353/udp 5353 8080 8443
 VOLUME ["/etc/acme-dns", "/var/lib/acme-dns"]
-ENTRYPOINT ["./acme-dns"]
-EXPOSE 53 80 443
-EXPOSE 53/udp
+CMD ["/acme-dns"]
```

Comment:
`ENTRYPOINT` allows for easy additional flags without needing to know the binary path.

Reply:
thanks. changed.